### PR TITLE
A3 Compat DO NOT MERGE

### PR DESCRIPTION
**Pull request by fhartwig**
My current idea how to solve https://tree.taiga.io/project/fisx-thentos/task/37 is that instead of putting the entire user resource URL in the `X-User-Path` header, we can just put the path component of the header. I've tried this on an older version of thentos and it seems to fix the problem. I haven't managed to test this with this version of thentos, though, since the email sending functionality seems to be broken, so I cannot create a user to log in with. (thentos crashes with `ActionErrorUnknown sendUserConfirmationMail: no such context`, where the argument to `context` seems to be `frontend_url`.

### CODE REVIEW

**Comment by fisx** [comment]
I think this is your bug:

```sh
$ git grep frontend_url
[...]
thentos-core/src/Thentos/Config.hs:  :*> \
  Just "Please go to {{frontend_url}}{{activation_path}} to confirm your account."
```

Regarding changing the `X-User-Path` header: This is likely to have deep implications in adhocracy.  I think for now we need to deal with absolute URIs both towards frontend and towards backend.  If you want to do something else, we should discuss this before you spend too much time on it.

**Comment by fhartwig** [comment]
Can you be more specific about the implications? As far as I can tell, the `X-User-Path` is only used for authentication in a3, and it seems like that handles both formats fine. I don't want to change anything that we return to the frontend. I've tested this now (the email issue was fixed in #396) and at least in my setup it seems to work. If you have any pointers where potential problems might be, that would be helpful.

File: thentos-adhocracy/src/Thentos/Adhocracy3/Backend/Api/Simple.hs [hunk 1]
```diff
@@ -612,17 +614,18 @@ a3RenderUser uid _ = externalUrlOfDefaultPersona uid
...
+    -- FIXME: get rid of the fromJust, a3-Prefix should not be optional
+    a3Prefix = fromJust $ config >>. (Proxy :: Proxy '["a3-prefix"])
 
 userIdFromPath :: MonadError (ThentosError e) m => Path -> m UserId
 userIdFromPath (Path s) = do
     uri <- either (const . throwError . MalformedUserPath $ s) return $
         URI.parseURI URI.laxURIParserOptions $ cs s
     rawId <- maybe (throwError $ MalformedUserPath s) return $
+        -- FIXME: I don't think this will work, as the path may be something
+        --  like /foo/principals/users/ (see https://github.com/liqd/adhocracy3/pull/1849)
```
Comment (fisx) [inline thentos-adhocracy/src/Thentos/Adhocracy3/Backend/Api/Simple.hs]: Then this should be fixed immediately, not left in as a FIXME, right?

Comment (fisx) [inline thentos-adhocracy/src/Thentos/Adhocracy3/Backend/Api/Simple.hs]: (I mean: it is a BUG, not a FIXME.)

File: thentos-adhocracy/src/Thentos/Adhocracy3/Backend/Api/Simple.hs [hunk 2]
```diff
@@ -612,17 +614,18 @@ a3RenderUser uid _ = externalUrlOfDefaultPersona uid
 -- | Convert a local file name into a absolute path relative to the A3 backend endpoint.  (Returns
 -- exposed url.)
 a3backendPath :: ThentosConfig -> ST -> Path
-a3backendPath config localPath = Path $ cs (exposeUrl beHttp) <//> localPath
+a3backendPath config localPath = Path $ a3Prefix <//> localPath
   where
-    beHttp     = case config >>. (Proxy :: Proxy '["backend"]) of
-                     Nothing -> error "a3backendPath: backend not configured!"
-                     Just v -> Tagged v
+    -- FIXME: get rid of the fromJust, a3-Prefix should not be optional
```
Comment (fisx) [inline thentos-adhocracy/src/Thentos/Adhocracy3/Backend/Api/Simple.hs]: TODO, not FIXME.

File: thentos-adhocracy/src/Thentos/Adhocracy3/Backend/Api/Simple.hs (hunk 1 above)
Comment (fhartwig) [inline thentos-adhocracy/src/Thentos/Adhocracy3/Backend/Api/Simple.hs]: Yes, I think this should be fixed soon, but it's out of scope for this PR. I'll mark it as BUG and open a ticket.

**Comment by fhartwig** [comment]
Closing in favour of #412

**Pull request closed without merging**